views/graph.py

```diff
@@ -64,76 +77,184 @@ def closeEvent(self, event):
+                for item in self.location_model.rows:
+                    if item.active.value:
+                        self.removeItem(item.plots(
+                                self.current_parameter['parameters'], i))
 
-    def data_changed(self, index):
+    def ds_data_changed(self, index):
+        """
+        change graphs based on changes in locations
+        :param index: index of changed field
+        """
+        if self.ds_model.columns[index.column()].name == 'active':
 
-        if self.model.columns[index.column()].name == 'active':
-            active = self.model.rows[index.row()].active.value
+            active = self.ds_model.rows[index.row()].active.value
 
             if active:
-                self.show_timeseries(index.row())
+                for i in range(0, len(self.location_model.rows)):
+                    if self.location_model.rows[i].active.value:
+                        self.show_timeseries(i, index.row())
             else:
-                self.hide_timeseries(index.row())
-        elif self.model.columns[index.column()].name == 'hover':
-            item = self.model.rows[index.row()]
+                for i in range(0, len(self.location_model.rows)):
+                    if self.location_model.rows[i].active.value:
+                        self.hide_timeseries(i, index.row())
+
+    def on_insert_locations(self, parent, start, end):
+        """
+        add list of items to graph. based on Qt addRows model trigger
+        :param parent: parent of event (Qt parameter)
+        :param start: first row nr
+        :param end: last row nr
+        """
+        for i in range(start, end+1):
+            item = self.location_model.rows[i]
+            for ds in self.ds_model.rows:
+                if ds.active.value:
+                    index = self.ds_model.rows.index(ds)
+                    self.addItem(item.plots(
+                            self.current_parameter['parameters'], index))
+
+    def on_remove_locations(self, index, start, end):
+        """
+        remove items from graph. based on Qt model removeRows
+        trigger
+        :param index: Qt Index (not used)
+        :param start: first row nr
+        :param end: last row nr
+        """
+        for i in range(start, end+1):
+            item = self.location_model.rows[i]
+            if item.active.value:
+                for ds in self.ds_model.rows:
+                    if ds.active.value:
+                        index = self.ds_model.rows.index(ds)
+                        self.removeItem(item.plots(
+                                self.current_parameter['parameters'], index))
+
+    def location_data_changed(self, index):
+        """
+        change graphs based on changes in locations
+        :param index: index of changed field
+        """
+        if self.location_model.columns[index.column()].name == 'active':
+            active = self.location_model.rows[index.row()].active.value
+
+            if active:

```

Comment:
De `if active` check kan beter helemaal binnen in de loop gedaan worden. De "for" (en de "if" erna) zijn voor allebei de versies hetzelfde.